plugins/linux-spi-lpc/fu-plugin-linux-spi-lpc.c

```diff
@@ -0,0 +1,122 @@
+	if (fu_common_is_cpu_intel ())
+		fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRED_FOR, "hsi");
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	if (!g_file_test (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, G_FILE_TEST_IS_DIR)) {
+		g_set_error (error,
+			     G_IO_ERROR,
+			     G_IO_ERROR_NOT_FOUND,
+			     "Cannot find %s",
+			     FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_add_security_attr_bioswe (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FwupdSecurityAttr *attr;
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *fn = NULL;
+
+	/* load file */
+	fn = g_build_filename (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, "bioswe", NULL);
+	if (!g_file_get_contents (fn, &buf, &bufsz, error))
+		return FALSE;
+	attr = fwupd_security_attr_new ("org.kernel.BIOSWE");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "SPI");
+	if (g_strcmp0 (buf, "0\n") == 0) {
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+		fwupd_security_attr_set_result (attr, "Write disabled");
+	} else {
+		fwupd_security_attr_set_result (attr, "Write enabled");
+	}
+	g_ptr_array_add (attrs, attr);
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_add_security_attr_ble (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FwupdSecurityAttr *attr;
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *fn = NULL;
+
+	/* load file */
+	fn = g_build_filename (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, "ble", NULL);
+	if (!g_file_get_contents (fn, &buf, &bufsz, error))
+		return FALSE;
+	attr = fwupd_security_attr_new ("org.kernel.BLE");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "SPI");
+	if (g_strcmp0 (buf, "1\n") == 0) {
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+		fwupd_security_attr_set_result (attr, "Lock enabled");
+	} else {
+		fwupd_security_attr_set_result (attr, "Lock disabled");
+	}
+	g_ptr_array_add (attrs, attr);
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_add_security_attr_smm_bwp (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FwupdSecurityAttr *attr;
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_autofree gchar *fn = NULL;
+
+	/* load file */
+	fn = g_build_filename (FU_PLUGIN_LINUX_SPI_LPC_SYSFS_DIR, "smm_bwp", NULL);
+	if (!g_file_get_contents (fn, &buf, &bufsz, error))
+		return FALSE;
+	attr = fwupd_security_attr_new ("org.kernel.SMM_BWP");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "SPI");
+	if (g_strcmp0 (buf, "1\n") == 0) {
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+		fwupd_security_attr_set_result (attr, "SMM required");
+	} else {
+		fwupd_security_attr_set_result (attr, "SMM not required");
+	}
```

Comment:
This stuff isn't mandating SMM, it's BIOS write protect

Reply:
Right, as I understand it it's write protect, unless in SMM mode -- right?

Reply:
It's locking down the region that the BIOS is stored to only SMM writes

Reply:
So for failure "Non-SMM writes permitted" and for success "SMM write only"?

Reply:
I think that's simplifying it too much.  How about this:
Title: `BIOS region of SPI`
Success: `Writes only available through BIOS code`
Fail: `Writes available to operating system`

Reply:
I've done something like that, thanks.